src/components/EndorsementChain/EndorsementChainLayout/EndorsementChainLayout.tsx

```diff
@@ -8,176 +8,239 @@ import { getLogger } from "../../../utils/logger";
         case "Burnt":
           tableRows.push(
-            <tr className="table-row" key={index++}>
-              <td className="table-cell date border-top-none">
-                {format(new Date(tradetrustErc721Event?.eventTimestamp ?? 0), "do MMM yyyy, hh:mm aa")}
-              </td>
-              <td className="table-cell endorsement-ui-dash border-top-none" colSpan={2}>
-                <div className="relative flex">
-                  <div className="dot" data-testid="dot" />
-                  <div className="name">Surrender of document accepted</div>
+            <tr className="table-row" key={tableRowIndex++}>
+              <td className="table-cell">
+                <div className="action-title">Surrender of document accepted</div>
+                <div className="date">
+                  {format(new Date(tradetrustErc721Event?.eventTimestamp ?? 0), "do MMM yyyy, hh:mm aa")}
                 </div>
               </td>
+              <td className="table-cell" colSpan={2}>
+                <EndorsementJourney
+                  displayDashHead={true}
+                  displayDot={true}
+                  displayDashTail={eventIndex + 1 !== endorsementChain.length}
+                />
+              </td>
             </tr>
           );
           break;
+
+        // ui for each transfer to wallet row
         case "Transfer to Wallet":
           tableRows.push(
-            <tr className="table-row" key={index++}>
-              <td className="table-cell date border-top-none">
-                {format(new Date(tradetrustErc721Event?.eventTimestamp ?? 0), "do MMM yyyy, hh:mm aa")}
-              </td>
-              <td className="table-cell endorsement-ui-dash border-top-none" colSpan={2}>
-                <div className="relative flex flex-col">
-                  <div className="dot" data-testid="dot" />
-                  <div className="name" data-testid="transferred-to-wallet">
-                    Transferred to wallet
-                  </div>
-                  <div className="address">{tradetrustErc721Event.documentOwner}</div>
+            <tr className="table-row" key={tableRowIndex++}>
+              <td className="table-cell">
+                <div className="action-title" data-testid="transferred-to-wallet">
+                  Transferred to wallet
                 </div>
+                <div className="date">
+                  {format(new Date(tradetrustErc721Event?.eventTimestamp ?? 0), "do MMM yyyy, hh:mm aa")}
+                </div>
+              </td>
+              <td className="table-cell">
+                <EndorsementJourney
+                  displayDashHead={true}
+                  displayDot={true}
+                  displayDashTail={eventIndex + 1 !== endorsementChain.length}
+                />
+                <div className="address">{tradetrustErc721Event.documentOwner}</div>
+              </td>
+              <td className="table-cell">
+                <EndorsementJourney
```

Comment:
i think Surrender, Burnt, Transfer to Wallet maybe can further extract out cos they look similar, so this new component only takes in action title as prop + journey as children

Reply:
ok i try extract it further